src/components/slider/slider.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  Component,
+  ElementRef,
+  HostBinding,
+  Input,
+  ViewEncapsulation,
+  AfterContentInit,
+} from '@angular/core';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-slider',
+  host: {
+    'tabindex': '0',
+    '(click)': 'onClick($event)',
+    '(drag)': 'onDrag($event)',
+    '(dragstart)': 'onDragStart($event)',
+    '(dragend)': 'onDragEnd()',
+    '(window:resize)': 'onResize()',
+    '(blur)': 'onBlur()',
+  },
+  templateUrl: 'slider.html',
+  styleUrls: ['slider.css'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class MdSlider implements AfterContentInit {
+  /** A renderer to handle updating the slider's thumb and fill track. */
+  private _renderer: SliderRenderer = null;
+
+  /** The dimensions of the slider. */
+  private _sliderDimensions: ClientRect = null;
+
+  /** Whether or not the slider is disabled. */
+  private _disabled: boolean = false;
+
+  /** The miniumum value that the slider can have. */
+  private _min: number = 0;
+
+  /** The maximum value that the slider can have. */
+  private _max: number = 100;
+
+  /** The percentage of the slider that coincides with the value. */
+  private _percent: number = 0;
+
+  /**
+   * Whether or not the thumb is currently being dragged.
+   * Used to determine if there should be a transition for the thumb and fill track.
+   * @internal
+   */
+  isDragging: boolean = false;
+
+  /**
+   * Whether or not the slider is active (clicked or is being dragged).
+   * Used to shrink and grow the thumb as according to the Material Design spec.
+   * @internal
+   */
+  isActive: boolean = false;
+
+  /** Indicator for if the value has been set or not. */
+  private _isValueInitialized: boolean = false;
+
+  /** Value of the slider. */
+  private _value: number = 0;
+
+  @Input()
+  @HostBinding('class.md-slider-disabled')
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value) {
+    this._disabled = (value != null && value !== false) ? true : null;
+  }
+
+  @Input()
+  @HostBinding('attr.aria-valuemin')
+  get min() {
+    return this._min;
+  }
+
+  set min(v: number) {
+    // This has to be forced as a number to handle the math later.
+    this._min = Number(v);
+
+    // If the value wasn't explicitly set by the user, set it to the min.
+    if (!this._isValueInitialized) {
+      this.value = this._min;
+    }
+  }
+
+  @Input()
+  @HostBinding('attr.aria-valuemax')
+  get max() {
+    return this._max;
+  }
+
+  set max(v: number) {
+    this._max = Number(v);
+  }
+
+  @Input()
+  @HostBinding('attr.aria-valuenow')
+  get value() {
+    return this._value;
+  }
+
+  set value(v: number) {
+    this._value = Number(v);
+    this._isValueInitialized = true;
```

Comment:
Rather than setting this here, you should set `_isInitialied = true` in `ngAfterViewInit` 
(pointed by by @DevVersion)